server/models/estimationFeatureModel.js

```diff
@@ -126,5 +126,79 @@ estimationFeatureSchema.statics.addFeatureByEstimator = async (featureInput, est
 }
 
 
+estimationFeatureSchema.statics.updateFeatureByEstimator = async (featureInput, estimator) => {
+    validate(featureInput, estimationEstimatorUpdateFeatureStruct)
+
+    if (!estimator || !userHasRole(estimator, SC.ROLE_ESTIMATOR))
+        throw new AppError('Not an estimator', EC.INVALID_USER, EC.HTTP_BAD_REQUEST)
+
+    let estimationFeature = await EstimationFeatureModel.findById(featureInput._id)
+    if(!estimationFeature)
+        throw new AppError('Estimation feature not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    let estimation = await EstimationModel.findById(estimationFeature.estimation._id)
+    if (!estimation)
+        throw new AppError('Estimation not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    if (!_.includes([SC.STATUS_ESTIMATION_REQUESTED, SC.STATUS_CHANGE_REQUESTED], estimation.status))
+        throw new AppError("Estimation has status as ["+estimation.status+"]. Estimator can only update feature into those estimations where status is in [" + SC.STATUS_ESTIMATION_REQUESTED + ", " + SC.STATUS_CHANGE_REQUESTED + "]", EC.INVALID_OPERATION, EC.HTTP_BAD_REQUEST)
+
+    if (estimationFeature.repo && estimationFeature.repo._id) {
+        featureInput.repo = {
+            _id: estimationFeature.repo._id
+        }
+        let updateFeatureFromRepo = await RepositoryModel.updateFeature(featureInput,estimator)
+    }
+
+    estimationFeature.technologies = featureInput.technologies
+    estimationFeature.tags = featureInput.tags
+
+    if (!_.isEmpty(featureInput.notes)) {
+        featureInput.notes = featureInput.notes.map(n => {
+            n.name = estimator.fullName
+            return n
+        })
+    }
+    let  mergeAllNotes = estimationFeature.notes
+    if (!_.isEmpty(mergeAllNotes)) {
+       featureInput.notes.map(n => {
+            mergeAllNotes.push(n)
+        })
+    }else{
+        mergeAllNotes = featureInput.notes
+    }
+    estimationFeature.notes = mergeAllNotes
```

Comment:
Name/Description of feature would be changed during an update if user has appropriate permissions. Update of name/description is missing.